lib/urldata.h

```diff
@@ -753,6 +753,7 @@ struct SingleRequest {
                         specific upload buffers. See readmoredata() in http.c
                         for details. */
   BIT(no_body);      /* the response has no body */
+  BIT(fread_func_eof); /* the read callback has returned EOF */
```

Comment:
are there any scenarios where this needs to cleared like code that calls the seek function

Reply:
Good thinking. I would say there is. I'll check what test coverage we have for this. Ideally this PR should fail a test where we rewind...

Reply:
The `SingleRequest` struct is cleared first thing in `Curl_connect` so it seems good.